taxi/views.py

```diff
@@ -79,6 +82,24 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:car-list")
 
 
+@login_required
```

Comment:
You can put this logic into post method of CarDetailView. This way you don't need to create extra views and url endpoints to handle adding/deleting driver.
You will need to update car_detail template to use form instead of hyperlink to send request to the CarDetailView to make this work.